pandas/core/indexes/datetimes.py

```diff
@@ -358,11 +358,6 @@ def __new__(cls, data=None,
                 msg = 'periods must be a number, got {periods}'
                 raise TypeError(msg.format(periods=periods))
 
-        if data is None and freq is None \
-                and com._any_none(periods, start, end):
-            raise ValueError("Must provide freq argument if no data is "
-                             "supplied")
```

Comment:
small refactor here; moved this code a few lines down so all the `if data is None:` code is in one place.